### PR TITLE
Allow non-interactive mode for trup_call commands

### PR DESCRIPTION
**Pull request by pdostal**
## What the pull request does?
 * Do not rely on `root-console` when `transactional::trup_call` command starts with `-n parameter`.
 * Switch to `root-console` in the beginning of `transactional::process_reboot` and back at the end.

## How will it help?
when we work serial terminal, we currently have to switch to `root-console`:
```perl
select_console 'root-console';
trup_call 'pkg install ed';
check_reboot_changes;
select_serial_terminal;
```
After this patch we will be able to do just continue in non-interactive mode:
```perl
trup_call '-n pkg install ed';
check_reboot_changes;
```

- Related pull request: #15846, #15814
- Verification run: [MicroOS](https://pdostal-server.suse.cz/tests/2916), [SLE Micro](https://pdostal-server.suse.cz/tests/2917)

### CODE REVIEW

File: lib/transactional.pm [hunk 1]
```diff
@@ -172,8 +179,9 @@ sub trup_call {
                 die 'Abort dialog not shown' unless wait_serial('Abort');
                 send_key 'ret';
             }
-        }
-        else {
+        } elsif ($cmd =~ /^-n/) {
```
Comment (Vogtinator) [inline lib/transactional.pm:182]: This should come before the `wait_serial` to not waste 30s waiting for something which doesn't happen. Especially with an automatic reboot at the end that might cause openQA to miss the reboot.

Comment (mloviska) [inline lib/transactional.pm:182]: This puts some restrictions on the caller as it is now required to use non-interactive as first argument

File: lib/transactional.pm [hunk 2]
```diff
@@ -102,6 +106,9 @@ sub process_reboot {
         select_console 'root-console';
         assert_script_run 'clear';
     }
+
+    # Switch to the previous console
+    ($prev_console) ? select_console($prev_console) : die('The $prev_console is not defined.');
```
Comment (mloviska) [inline lib/transactional.pm:111]: Is this necessary? Is there any situation that `current_console` returns a falsy value?

Comment (pdostal) [reply lib/transactional.pm:111]: It's not. I'll get rid of it.

File: lib/transactional.pm (hunk 1 above)
Comment (pdostal) [reply lib/transactional.pm:182]: Good point!

Comment (pdostal) [reply lib/transactional.pm]: From the `transactional-update --help` output:
```bash
# transactional-update --help
...
Syntax: transactional-update [option...] [general-command...] [package-command]
        transactional-update [option...] standalone-command
…
Options:
--interactive, -i          Use interactive mode for package command
--non-interactive, -n      Use non-interactive mode for package command
```

My new regular expression:
```perl
$cmd =~ /-\w*n\w* pkg/
```

Samples I was testing on:
```bash
-n pkg install q # Valid
-q -n pkg install q # Valid
-qn pkg install q # Valid
-nq pkg install q # Valid
reboot -nq pkg install q # Valid
pkg -n install -q # Not valid
-q pkg -n install q # Not valid
```

File: lib/transactional.pm [hunk 3]
```diff
@@ -165,15 +172,16 @@ sub trup_call {
     $script .= "; echo trup-\$?- | tee -a /dev/$serialdev" unless $cmd =~ /reboot / && $args{exit_code} == 0;
     script_run $script, 0;
     if ($cmd =~ /pkg |ptf /) {
-        if (wait_serial "Continue?") {
+        if ($cmd =~ /(^|\s)-\w*n\w* pkg/) {
+            record_info 'non-interactive', 'The transactional-update command is in non-interactive mode';
+        } elsif (wait_serial "Continue?") {
```
Comment (foursixnine) [inline lib/transactional.pm:177]: I wish we could add a unit test for this kind of thing by extracting it.

**Review by foursixnine — APPROVED** [review]

File: lib/transactional.pm (hunk 3 above)
Comment (pdostal) [reply lib/transactional.pm:177]: I don't understand :/

Comment (foursixnine) [reply lib/transactional.pm:177]: > I don't understand :/

I mean, if we had the condition, in a sort routine, we could easily test the code, like we do for `version_check` for instance but we can think about this in the following weeks (I'm just making a note here)

**Comment by pdostal** [comment]
@mloviska @Vogtinator please re-check. This should be now ready to merge.

**Review by mloviska — APPROVED** [review]

**Pull request merged**